AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/Sync/RemoteSyncEngine+IncomingEventReconciliationQueueEvent.swift

```diff
@@ -0,0 +1,56 @@
+//
+// Copyright 2018-2020 Amazon.com,
+// Inc. or its affiliates. All Rights Reserved.
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+import Amplify
+import Combine
+import Foundation
+
+@available(iOS 13.0, *)
+extension RemoteSyncEngine {
+    @available(iOS 13.0, *)
+    func onReceiveCompletion(receiveCompletion: Subscribers.Completion<DataStoreError>) {
+        let semaphore = DispatchSemaphore(value: 1)
+        semaphore.wait()
```

Comment:
guessing this gets called multiple times and you only want it called once? can you removeDuplicates() on the sink that calls this

Reply:
Great callout -- and actually this may not be needed at all

What I was actually trying to protect against is not multiple calls on a single subscription - but I think there was a bug in my clean up logic, where i was only pausing the queue (rather than destroying it), which lead to multiple onReceiveCompletions.